cmd/state-svc/main.go

```diff
@@ -48,13 +48,21 @@ func main() {
 		os.Exit(exitCode)
 	}()
 
+	cfg, err := config.New()
+	if err != nil {
+		// We do not want to log an error here as we want to avoid potential rollbar reports until we load the config
+		logging.Debug("Failed to load configuration: %v", err)
+	} else {
+		logging.CurrentHandler().SetConfig(cfg)
+	}
```

Comment:
This is a little bit awkward. Most of the run functions also initialize at config instance and handle the error returned from `config.New()`. We need the config to determine if we should send rollbar reports so we initialize it here before the run function and only debug log the error, leaving the run function to also initialize a config instance and handle the error.

Reply:
I think this means we need to initialize cfg earlier in the process rather than in the `run()` function.

Reply:
There is a bit of a chicken and egg problem here. Being unable to load the config would be an error we would want to get via rollbar but we need to load the config in order to determine if we want to send rollbar reports in the first place. I've left the config initialization in the `run()` function as we're only debug logging the error here.

Reply:
I don't think that matters, unless I'm missing something. If it can't initialize config then it can report it to rollbar, and then error out. My comment was more about only initializing config once, and then passing it around. If it failed then it wouldn't pass it around cause it would've errored out anyway.